### PR TITLE
Document Functions and Types

### PR DESCRIPTION
**Pull request by CodaFi**
Gonna split these up in the next one

### CODE REVIEW

File: Sources/LLVMSwift/LLVMType.swift [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
+  ///
+  /// The `undef` value can be used anywhere a constant is expected, and
+  /// indicates that the user of the value may receive an unspecified 
+  /// bit-pattern.
   public func undef() -> IRValue {
     return LLVMGetUndef(asLLVM())
   }
-  
+
+  /// Returns the special LLVM constant `null` pointer value for this type.
```
Comment (harlanhaskins) [inline Sources/LLVMSwift/LLVMType.swift]: I think this should be stated as a pointer to the type initialized to null.

File: Sources/LLVMSwift/LLVMType.swift [hunk 2]
```diff
@@ -74,89 +85,159 @@ internal func convertType(_ type: LLVMTypeRef) -> IRType {
...
     return null()
   }
-  
-  public func constant<IntTy: Integer>(_ value: IntTy, signExtend: Bool = false) -> LLVMValueRef {
+
+  /// Creates an integer constant value with the given Swift integer value.
+  ///
+  /// - parameter value: A Swift integer value.
+  /// - parameter signExtend: Whether to sign-extend this value to fit this
+  ///   type's bit width.  Defaults to `false.
```
Comment (harlanhaskins) [inline Sources/LLVMSwift/LLVMType.swift]: Missing `

**Review by harlanhaskins — APPROVED** [review]
LGTM modulo nitpicks :shipit:

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**